aws/data_source_aws_ecr_lifecycle_policy_document.go

```diff
@@ -0,0 +1,201 @@
+
+		Schema: map[string]*schema.Schema{
+			"rule": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"priority": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Required: false,
+							Optional: true,
+						},
+						"selection": {
+							Type:     schema.TypeSet,
+							Required: false,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"tag_status": {
+										Type:         schema.TypeString,
+										Required:     false,
+										Optional:     true,
+										Default:      "any",
+										ValidateFunc: validation.StringInSlice([]string{"tagged", "untagged", "any"}, false),
+									},
+									"tag_prefix_list": {
+										Type:     schema.TypeList,
+										Required: false,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+									"count_type": {
+										Type:         schema.TypeString,
+										Required:     false,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"imageCountMoreThan", "sinceImagePushed", "any"}, false),
+									},
+									"count_unit": {
+										Type:         schema.TypeString,
+										Required:     false,
+										Optional:     true,
+										ValidateFunc: validation.StringInSlice([]string{"days"}, false),
+									},
+									"count_number": {
+										Type:     schema.TypeInt,
+										Required: false,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"action": {
+							Type:     schema.TypeSet,
+							Required: false,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  "expire",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"json": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEcrLifecyclePolicyDocumentRead(d *schema.ResourceData, meta interface{}) error {
+	mergedDoc := &EcrLifecyclePolicyDoc{}
+
+	// process the current document
+	doc := &EcrLifecyclePolicyDoc{}
+
+	var cfgStmts = d.Get("rule").([]interface{})
+	stmts := make([]*EcrLifecyclePolicyStatement, len(cfgStmts))
+	for i, stmtI := range cfgStmts {
+		cfgStmt := stmtI.(map[string]interface{})
+
+		stmt := &EcrLifecyclePolicyStatement{
+			RulePriority: cfgStmt["priority"].(int),
+		}
+
+		if description, ok := cfgStmt["description"]; ok {
+			stmt.Description = description.(string)
+		}
+
+		if selection := cfgStmt["selection"].(*schema.Set).List(); len(selection) > 0 {
+			stmt.Selection = dataSourceAwsEcrLifecyclePolicyDocumentMakeSelection(selection)
+		}
+
+		stmt.Action = EcrLifecyclePolicyAction{
```

Comment:
This should either attempt to read `cfgStmt["action"]` (and default to adding this configuration if its length is 0) or the `action` attribute should be removed.

Reply:
I think I follow.  Is this (1a4f6cc) what you are looking for here?

Reply:
Yes, thank you!